Update and clarify archival process (#1050)
cncf/toc: process/archiving.md

* Adoption stats

Automated systems and the references above can trigger a review by the TOC. For example a project might be very mature and close to feature complete, which would be different from general inactiveness. The archival process itself remains manual with close attention to the details by TOC.

## Archiving Process

This is a multi-step process with ample time for projects, adopters, and community members to react, discuss, and develop improvements. As a result, it may take an unspecified time. CNCF TOC's intent is to be deliberate in understanding the state of the project and if the project wants and can to return to an active state.

The process can be stopped by TOC consensus at any time until a formal vote has been announced.

1. Archival review is triggered (see above).
2. TOC opens an issue inquiring about the health of the project. While there's not fixed timeline, it will be in the order of months.
3. A formal archival request is created in https://github.com/cncf/toc/issues and the issue tracker of the member project, cross-linking both.
4. TOC informs project maintainers and the end user community. This step will usually be delegated to CNCF staff.

> 5. Once enough information has been provided, CNCF TOC may announce the mandatory two week public comment period

Author: ```suggestion
5. Once enough information has been provided, CNCF TOC may announce a mandatory public comment period of two weeks. This period may be extended by TOC as needed.
```

* Project meetings schedules and notes
* Public mailing list activity
* Adoption stats

Automated systems and the references above can trigger a review by the TCO, for example a project might be very mature and close to feature complete, which would be different from general inactiveness. The archival process itself remains manual with close attention to the details by TOC.

## Archiving Process

This is a multi-step process with ample time for projects to react, discuss, and improve built in. Being slow and deliberate is a bug, not a feature. The process can be stopped by TOC consensus at any time until a formal vote has been announced.

1. Archiving review is triggered (see above).
2. TOC opens an issue inquiring about the health of the project. While there's not fixed timeline, it will be in the order of months.
3. A formal archival request is created in https://github.com/cncf/toc/issues and the issue tracker of the member project, cross-linking both.
4. TOC informs project maintainers and the end user community. This step will usually be delegated to CNCF staff.
5. A TOC vote is announced; the voting duration is two weeks. A 2/3 majority in favor of archiving is required.

> ## Archiving Status

Author: ```suggestion
## Archived Status
```

Open source projects have a lifecycle and there are times that projects become inactive due to a variety of reasons. There are also cases where a project may no longer want to be supported by the TOC, or the TOC may no longer wish to recommend the use of a project.

> ## Triggering Archiving review 

Reviewer: ```suggestion
## Triggering a Review to Archive 
```

CNCF TOC may also look at a variety of signals, including but not limited to:
* [CNCF Project health dashboard](https://projecthealth.cncf.io/)
* If the project would currently meet criteria for acceptance to CNCF

> * Requests for archival by anyone in the wider CNCF community

Reviewer: ```suggestion
* Requests to archive by anyone in the wider CNCF community
```

* Security posture of the project, in particular speedy reaction to and resolution of vulnerability reports
* Project meetings schedules and notes
* Public mailing list activity
* Adoption stats

> Automated systems and the references above can trigger a review by the TCO, for example a project might be very mature and close to feature complete, which would be different from general inactiveness. The archival process itself remains manual with close attention to the details by TOC.

Reviewer: ```suggestion
Automated systems and the references above can trigger a review by the TOC. For example a project might be very mature and close to feature complete, which would be different from general inactiveness. The archival process itself remains manual with close attention to the details by TOC.
```

## Archiving Process

> This is a multi-step process with ample time for projects to react, discuss, and improve built in. Being slow and deliberate is a bug, not a feature. The process can be stopped by TOC consensus at any time until a formal vote has been announced.

Reviewer: ```suggestion
This is a multi-step process with ample time for projects, adopters, and community members to react, discuss, and develop improvements. As a result, these may take time however the intent is to be deliberate in understanding the state of the project and if it can or wants to return to an active state. The process can be stopped by TOC consensus at any time until a formal vote has been announced.
```

Author: What about:

```suggestion
This is a multi-step process with ample time for projects, adopters, and community members to react, discuss, and develop improvements. As a result, it may take an unspecified time. CNCF TOC's intent is to be deliberate in understanding the state of the project and if the project wants and can to return to an active state.

The process can be stopped by TOC consensus at any time until a formal vote has been announced.
```

> 1. Archiving review is triggered (see above).

Reviewer: ```suggestion
1. Archival review is triggered (see above).
```

2. TOC opens an issue inquiring about the health of the project. While there's not fixed timeline, it will be in the order of months.
3. A formal archival request is created in https://github.com/cncf/toc/issues and the issue tracker of the member project, cross-linking both.
4. TOC informs project maintainers and the end user community. This step will usually be delegated to CNCF staff.

> 5. A mandatory two week discussion phase begins

Reviewer: ```suggestion
5. Once enough information has been provided, the TOC will kick off a mandatory two week discussion
```

Author: ```suggestion
5. Once enough information has been provided, the TOC may announce the mandatory two week public discussion
```

Author: ```suggestion
5. Once enough information has been provided, CNCF TOC may announce the mandatory two week public comment period
```

> 5. A mandatory two week discussion phase begins

Reviewer 2: nit: s/discussion/public comment period

to stay consistent with nomenclature with other TOC processes

Author: Good catch, will update, thanks!

> 6. A TOC vote is announced; the voting duration is two weeks. A 2/3 majority in favor of archiving is required.

Reviewer: ```suggestion
6. Upon conclusion, a TOC vote is announced; the voting duration is two weeks. A 2/3 majority in favor of archiving is required.
```

Reviewer: LGTM

## Archived Status

What does archiving for a CNCF project mean?

* CNCF will no longer provide support for the project via service desk
* CNCF will list the archived project at https://www.cncf.io/archived-projects/
* Trademarks and domain names of archived projects are still hosted by the CNCF and the Linux Foundation
* CNCF can provide services such as documentation updates to help transition users.
* Other CNCF marketing activities will no longer be provided for the project

> ## Reactivating an Archived Project

Reviewer 3: Can it be made explicit here that -  project source code repositories will continue to be maintained by the respective project community and CNCF will not force them to make the project read-only.

Author: Good point. Like so?

https://github.com/cncf/toc/pull/1050/commits/75fac3d892412fb4c7a2829d40b09d6ac5d6c2ab

> Any project can be reactivated into CNCF by following the normal [project proposal](https://github.com/cncf/toc/blob/main/process/project_proposals.md) and/or [sandbox](https://github.com/cncf/toc/blob/main/process/sandbox.md) process.

Reviewer 2: If any Incubating project is Archived and they want to be reactivated, would this mean that they would need to reapply as a Sandbox project or can reactivate as an Incubating project directly?

Reviewer: Nothing should prevent a project from reapplying back to the level they were archived at, so long as they meet the criteria for acceptance.

Author: Agreed. Also see https://github.com/cncf/toc/pull/1050/commits/75fac3d892412fb4c7a2829d40b09d6ac5d6c2ab
